src/ships.js

```diff
@@ -0,0 +1,36 @@
+class Ship {
+    constructor(itinerary) {
+
+        this.itinerary = itinerary;
+        this.currentPort = itinerary.ports[0];
+        this.previousPort = null;
+        this.currentPort.addShip(this);
+    };
+
+    setSail() {
+        this.currentPort.removeShip(this);
+
+        const itinerary = this.itinerary;
```

Comment:
No need to reassign here, you can call it directly this.itinerary.ports ... in the next const.